cli/src/commands/config/edit.rs

```diff
@@ -41,5 +42,17 @@ pub fn cmd_config_edit(
         file.save()?;
     }
     editor.edit_file(file.path())?;
+
+    // Trying to load back config. If error, saves the old config and warn the user.
+    if let Err(e) = ConfigLayer::load_from_file(file.layer().source, file.path().to_path_buf()) {
+        writeln!(
+            ui.error_with_heading("Error: "),
+            "An error has been found inside your config. Restoring previous one."
+        )?;
+
+        // Saving back previous config
+        file.save()?;
```

Comment:
We'll probably need to copy the bad file to temporary file. The user wouldn't want to start editing from scratch.

Reply:
it would mean the user would have to manually copy from the temp file to the "true" config file, idk if it's something wanted 🤔

What about saving the old "working" config in a temp file, and if the actual config file don't work then we show a warning and load the last working config from this temp file? So the user could simply do `jj config edit` to edit the non-working config and as soon as it works jujutsu stars re-using it ?

Reply:
> Seems like a good solution to simply prompt the user to keep editing, it's simpler

Yes.